pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -281,15 +286,18 @@ func (pl *PodTopologySpread) calPreFilterState(ctx context.Context, pod *v1.Pod)
 
 			pair := topologyPair{key: c.TopologyKey, value: node.Labels[c.TopologyKey]}
```

Comment:
What if there's more than one constraint that has the same `TopologyKey`? 
We must eliminate the duplication otherwise `TpPairToMatchNum` would be wrong since we could count pods on the same node more than once.

Reply:
> Btw, are there any use cases for having two constraints with the same topologyKey?

I guess so? Like the app team wants to spread their application like this, but the platform team wants to spread all applications like that (via the mutation webhook). 
Anyways, if we cannot 100% say there isn't at all, and also given we don't prohibit it from having multiple entries with the same topologyKey, we cannot break this behaviour.

But,

> It only takes into consideration the last counts for given topologyKey for a node.

yeah... So, if there's multiple entries with the same topologyKey but with a different label selector, the count could be different, which shouldn't overwrite one another.. I think we should regard it as a bug, given, again, we don't have a limitation for a duplicated topologyKey.

Reply:
Can I make the fix in this PR or we want it to be cherry-pickable?

Reply:
I _think_ it deserves the cherry-pick; we should make a minimum change to fix the bug in another PR and cherry-pick it to past releases.

@kubernetes/sig-scheduling-leads what do you think? 
Always, I am not sure about this kind of cherry-pick decision, the following instruction is vague.

> Critical bug fixes (loss of data, memory corruption, panic, crash, hang)
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md